### PR TITLE
Use Docsy-friendly layout for community page

### PR DESCRIPTION
**Pull request by sftim**
Adapt the existing page to work with (near-)vanilla Docsy.

Intended to help with https://github.com/kubernetes/website/issues/41171.

/area web-development

[Preview](https://deploy-preview-47415--kubernetes-io-main-staging.netlify.app/community/) | [existing](https://k8s.io/community)

### CODE REVIEW

**Comment by sftim** [comment]
BTW, the logo for 𝕏 isn't quite right. I'd be happy to have that fixed up in a follow up PR.

File: assets/scss/_custom.scss
Comment (dipesh-rawat) [inline assets/scss/_custom.scss:1]: A broken link is introduced as part of theses change while linking to the contribution guidelines _(reference below)_. We should address this link issue here.

> You can find out how to contribute to these docs in our [Contribution Guidelines](https://deploy-preview-47415--kubernetes-io-main-staging.netlify.app/docs/contribution-guidelines/)

**Review by dipesh-rawat — COMMENTED** [review]
Overall, the changes in this PR look great! There is potential for cleaning up the wording in the new top links section, which could be addressed separately by updating the [hugo.toml](https://github.com/kubernetes/website/blob/44669d56d7329c0a12eeaac31318f3025a91e51e/hugo.toml#L243-L292).

File: assets/scss/_custom.scss
Comment (dipesh-rawat) [reply assets/scss/_custom.scss:1]: Looks like the Docsy (the older version we're using) doesn't allow for configuring the contribution guide hyperlink and always defaults to `/docs/contribution-guidelines/`. As a quick fix, we could consider adding a redirect for now.

**Comment by sftim** [comment]
Revised; it's a bigger PR, but I like this better.

**Comment by sftim** [comment]
BTW: if we don't like the new initial page section, I recommend turning it off. But leave everything in this PR as is, because our overall aim is around alignment with Docsy (tweaking the community page is just a step on this journey).

In other words, if we don't like the new look, I'd add a one-line commit that turns off the new thing. And we'd be closer to switching to drop a lot of our Docsy customizations.

File: hugo.toml [hunk 1]
```diff
@@ -127,6 +127,7 @@ id = "UA-00000000-0"
 copyright_k8s = "The Kubernetes Authors"
 copyright_linux = "Copyright © 2020 The Linux Foundation ®."
 
+contribution_guidelines_url = "https://www.kubernetes.dev/docs/comms/"
```
Comment (dipesh-rawat) [inline hugo.toml:130]: Either we change this link or override the Docsy's default hyperlink text [`community_guideline`](https://github.com/google/docsy/blob/ab1f1d0257a09cce2fd820665b1b4900bf38921c/i18n/en.toml#L79) in our [en.toml](https://github.com/kubernetes/website/blob/main/data/i18n/en/en.toml) to accurately reflect where the link leads.

 > // Docsy’s [English-language strings](https://github.com/google/docsy/blob/main/i18n/en.toml)
[community_guideline]
other = "Contribution Guidelines"

**Review by dipesh-rawat — COMMENTED** [review]
Overall, I like the updated look of the [page](https://deploy-preview-47415--kubernetes-io-main-staging.netlify.app/community/) and the improved alignment with Docsy. 

There are couple of minor obeservations: 
- The new initial section could use some adjustment to be more centrally aligned.

 - The text `You can also read about our Contribution Guidelines.` currently links to a [page](https://www.kubernetes.dev/docs/comms/) with community communication platform guidelines, so a slight tweak there might be helpful _(added a suggestion below)_.

File: hugo.toml (hunk 1 above)
Comment (sftim) [reply hugo.toml:130]: I think what we really, really need are some actual contribution guidelines on https://k8s.dev/

“Implementing these is left as an exercise for the reader”

Comment (sftim) [reply hugo.toml:130]: Maybe https://www.kubernetes.dev/docs/contributor-cheatsheet/ is a better place to link to?

Comment (dipesh-rawat) [reply hugo.toml:130]: > Maybe https://www.kubernetes.dev/docs/contributor-cheatsheet/ is a better place to link to?

Agree, this link (https://www.kubernetes.dev/docs/contributor-cheatsheet/) will be an improvement over the current one. Additionally, adding a comment to localize the string `community_guideline` would be helpful too.

**Comment by dipesh-rawat** [comment]
> what's the minimal change I can make to get there?

Once the `contribution_guidelines_url` link is updated as proposed in the https://github.com/kubernetes/website/pull/47415#discussion_r1716156272, I think this will be ready for merging. Further refinements and improvements can be made later on separate from this PR.

**Comment by reylejano** [comment]
I agree, the changes look good
/approve

**Pull request merged**